src/main/java/org/jvnet/hudson/plugins/port_allocator/PortAllocator.java

```diff
@@ -36,14 +40,21 @@ private PortAllocator(PortType[] ports){
         this.ports = ports;
     }
 
+    @DataBoundConstructor
+    public PortAllocator(String pool) {
```

Comment:
I can confirm this is working. We have a total of 60 build jobs in our company, and we still have approx. 20 jobs that haven't been migrated to a Pipeline job structure yet. Both types of jobs work perfectly together, no issues w.r.t. co-existence or build failures after plugin upgrade.